core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -211,6 +211,32 @@ private[spark] class AppStatusListener(
           update(rdd, now)
         }
       }
+      // remove partition information
+      liveRDDs.values.foreach { rdd =>
+        rdd.getPartitions.values.foreach { partition =>
+          partition.executors.foreach { exec =>
+            if (exec.equals(event.executorId)) {
```

Comment:
You can use `filter` and avoid one level of indentation. Also you can use `==` in Scala.

In fact you can avoid two levels of indentation if you filter the partition list on whether `p.executors.contains(event.executorId)`.

And that can simplify solving the other problem in your code.